sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.institutions.getAssessmentAwardDetails.e2e-spec.ts

```diff
@@ -0,0 +1,263 @@
+          { disbursementReceipt },
+        ),
+      );
+    });
+    await db.disbursementReceiptValue.save(disbursementReceiptsValues);
+
+    const finalAwards = {};
+    finalAwards[`disbursementReceipt1Id`] = disbursementReceipt.id;
+
+    disbursementReceiptsValues.forEach((disbursementReceiptsValue) => {
+      finalAwards[
+        `disbursementReceipt1${disbursementReceiptsValue.grantType.toLowerCase()}`
+      ] = disbursementReceiptsValue.grantAmount;
+    });
+
+    const endpoint = `/institutions/assessment/student/${student.id}/assessment/${assessment.id}/award`;
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        applicationNumber: application.applicationNumber,
+        applicationStatus: application.applicationStatus,
+        institutionName:
+          assessment.offering.educationProgram.institution.operatingName,
+        offeringIntensity: assessment.offering.offeringIntensity,
+        offeringStudyStartDate: getDateOnlyFormat(
+          assessment.offering.studyStartDate,
+        ),
+        offeringStudyEndDate: getDateOnlyFormat(
+          assessment.offering.studyEndDate,
+        ),
+        estimatedAward: {
+          disbursement1Date: getDateOnlyFormat(
+            firstDisbursementSchedule.disbursementDate,
+          ),
+          disbursement1Status:
+            firstDisbursementSchedule.disbursementScheduleStatus,
+          disbursement1COEStatus: firstDisbursementSchedule.coeStatus,
+          disbursement1MSFAANumber: MASKED_MSFAA_NUMBER,
+          disbursement1MSFAAId: firstDisbursementSchedule.msfaaNumber.id,
+          disbursement1MSFAACancelledDate:
+            firstDisbursementSchedule.msfaaNumber.cancelledDate,
+          disbursement1MSFAADateSigned:
+            firstDisbursementSchedule.msfaaNumber.dateSigned,
+          disbursement1TuitionRemittance:
+            firstDisbursementSchedule.tuitionRemittanceRequestedAmount,
+          disbursement1Id: firstDisbursementSchedule.id,
+          disbursement1DocumentNumber: firstDisbursementSchedule.documentNumber,
+          ...("disbursement1bcsl" in awards && {
+            disbursement1bcsl: awards.disbursement1bcsl,
+          }),
+          ...("disbursement1csgf" in awards && {
+            disbursement1csgf: awards.disbursement1csgf,
+          }),
+          ...("disbursement1cslf" in awards && {
+            disbursement1cslf: awards.disbursement1cslf,
+          }),
+        },
+        finalAward: finalAwards,
+      });
+  });
+
+  it("Should throw not found exception when assessment is not found.", async () => {
+    // Arrange
+
+    // Student.
+    const student = await saveFakeStudent(db.dataSource);
+    const currentMSFAA = createFakeMSFAANumber(
```

Comment:
Does the assessment need an MSFAA?

Reply:
the controller will throw the same if the MSFAA is not there due to the inner join, I added the MSFAA here because, as part of this TC, I wanted to test that even if the MSFAA is there and the assessment doesn't exist, the controller is throwing `NOT_FOUND` error.